onnxruntime/core/session/standalone_op_invoker.cc

```diff
@@ -75,12 +75,13 @@ namespace standalone {
 void ReleaseNode(onnxruntime::Node* node) {
   if (node) {
     for (auto* input_arg : node->InputDefs()) {
-      delete input_arg;
+      std::unique_ptr<const onnxruntime::NodeArg>{input_arg};
```

Comment:
I think we need to fix the memory mgmt here instead of forcing a unique_ptr. In CreateOp(), before the Node is created if any exception is thrown we'll have a memory leak since the vectors of NodeArg* are not going to be deleted. Can we instead store the NodeArgs* as vector<unique_ptr<NodeArg>> in NodeRepo (which is in some sense analogous to Graph) keyed by (perhaps) the OpKernel* and then remove this entry in ReleaseOp? This way we won't need ReleaseNode at all.

Reply:
If we have exception during creation of Node in line 391, kernel will hence not be created and returned. So the "keyed by ... the OpKernel*" might not work.

Reply:
If you create a ```vector<unique_ptr<NodeArg>>``` before Node creation, even if an exception is thrown there's no leak + there's no need to save these in NodeRepo. These can be saved in NodeRepo at the end after a successful call to AddNode.

Reply:
Saved args firstly into a local vector and then node repo - preventing leaks on exceptions and ensuring the final release.